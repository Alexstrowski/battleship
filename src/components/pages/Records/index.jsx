import Board from '../Home/Board';

const ships = window.localStorage.getItem('boardHistory');
const boardHistory = ships ? JSON.parse(ships) : [];
console.log(boardHistory);
const Records = () => {
    return (
        <div className="container mx-auto">
            {boardHistory.map((ship, key) => {
                return (
                    <div className="flex justify-center" key={key}>
                        <Board isGameStarted={false} ships={ship} isRecord />
                    </div>
                );
            })}
        </div>
    );
};

export default Records;
