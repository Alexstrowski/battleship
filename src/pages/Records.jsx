import RecordsComponent from 'components/pages/Records';

const Home = () => {
    return <RecordsComponent />;
};

export default Home;
