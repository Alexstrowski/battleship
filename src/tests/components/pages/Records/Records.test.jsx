import { shallow, mount, render } from 'enzyme';
import Records from 'components/pages/Records';

describe('Tests in <Records />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Records />);
    });

    test('Must render <Records /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
