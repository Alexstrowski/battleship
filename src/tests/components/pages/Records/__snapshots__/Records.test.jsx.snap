// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Records /> Must render <Records /> correctly 1`] = `
<div
  className="container mx-auto"
/>
`;
